shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
+        Coding coding = getMSH9Coding(bundle);
+        coding.setDisplay(value);
+    }
+
+    // PID - Patient
+    public static Patient getPIDPatient(Bundle bundle) {
+        return resourceInBundle(bundle, Patient.class);
+    }
+
+    public static Patient createPIDPatient(Bundle bundle) {
+        Patient patient = new Patient();
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(patient));
+        return patient;
+    }
+
+    // PID-3 - Patient Identifier List
+    public static Identifier getPID3Identifier(Bundle bundle) {
+        Patient patient = getPIDPatient(bundle);
+        if (patient == null) {
+            return null;
+        }
+        return patient.getIdentifierFirstRep();
+    }
+
+    public static void setPID3Identifier(Bundle bundle, Identifier identifier) {
+        Patient patient = getPIDPatient(bundle);
+        if (patient == null) {
+            return;
+        }
+        patient.setIdentifier(List.of(identifier));
+    }
+
+    // PID-3.4 - Assigning Authority
+    public static Identifier getPID3_4Identifier(Bundle bundle) {
+        Identifier identifier = getPID3Identifier(bundle);
+        if (identifier == null) {
+            return null;
+        }
+        Organization organization = (Organization) identifier.getAssigner().getResource();
+        return organization.getIdentifierFirstRep();
+    }
+
+    public static void setPID3_4Identifier(Bundle bundle, Identifier identifier) {
+        Identifier patientIdentifier = getPID3Identifier(bundle);
+        if (patientIdentifier == null) {
+            return;
+        }
+        Organization organization = new Organization();
+        organization.setIdentifier(List.of(identifier));
+        Reference orgReference = createOrganizationReference(bundle, organization);
+        patientIdentifier.setAssigner(orgReference);
+    }
+
+    public static String getPID3_4Value(Bundle bundle) {
+        Identifier identifier = getPID3_4Identifier(bundle);
+        if (identifier == null) {
+            return null;
+        }
+        return identifier.getValue();
+    }
+
+    public static void setPID3_4Value(Bundle bundle, String value) {
+        Identifier identifier = getPID3_4Identifier(bundle);
+        if (identifier == null) {
+            return;
+        }
+        identifier.setValue(value);
+    }
+
+    // PID-3.5 - Identifier Type Code
+    public static Coding getPID3_5Coding(Bundle bundle) {
+        Identifier identifier = getPID3Identifier(bundle);
+        if (identifier == null) {
+            return null;
+        }
+        return identifier.getType().getCodingFirstRep();
+    }
+
+    public static void setPID3_5Coding(Bundle bundle, Coding coding) {
+        Identifier identifier = getPID3Identifier(bundle);
+        if (identifier == null) {
+            return;
+        }
+        identifier.setType(new CodeableConcept().addCoding(coding));
+    }
+
+    public static String getPID3_5Value(Bundle bundle) {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.